blockchain/validate.go

```diff
@@ -375,6 +376,34 @@ func checkProofOfWork(header *wire.BlockHeader, powLimit *big.Int, flags Behavio
 				" expected max of %064x", hashNum, target)
 			return ruleError(ErrHighHash, str)
 		}
+
+		err := validateEquihashSolution(header)
+
+		if err != nil {
+			return ruleError(ErrInvalidEquihash, "block has incorrect equihash solution")
+		}
+	}
+
+	return nil
+}
+
+//TODO: (siy) pass net parameters as parameter instead of using hardcoded mainnet
+// Validate equihash solution for given block
+func validateEquihashSolution(header *wire.BlockHeader) error {
+	headerBytes, err := header.SerializeAllHeaderBytes()
+	if err != nil {
+		return ruleError(ErrInvalidEquihash, "unable to deserialize required header fields")
+	}
+
+	result := equihash.ValidateEquihash(chaincfg.MainNetParams.N, chaincfg.MainNetParams.K, headerBytes,
+		int64(header.Nonce), header.EquihashSolution[:])
+
+	if err != nil {
```

Comment:
Is this error check needed? It seems redundant. It isn't updated since last check so there's no need to recheck.